src/packaging/tags.py

```diff
@@ -473,6 +473,58 @@ def mac_platforms(
                 yield f"macosx_{major_version}_{minor_version}_{binary_format}"
 
 
+def ios_platforms(
+    version: AppleVersion | None = None, multiarch: str | None = None
+) -> Iterator[str]:
+    """
+    Yields the platform tags for an iOS system.
+
+    :param version: A two-item tuple specifying the iOS version to generate
+        platform tags for. Defaults to the current iOS version.
+    :param multiarch: The CPU architecture+ABI to generate platform tags for -
+        (the value used by `sys.implementation._multiarch` e.g.,
+        `arm64_iphoneos` or `x84_64_iphonesimulator`). Defaults to the current
+        multiarch value.
+    """
+    if version is None:
+        # if iOS is the current platform, ios_ver *must* be defined. However,
+        # it won't exist for CPython versions before 3.13, which causes a mypy
+        # error.
+        _, release, _, _ = platform.ios_ver()  # type: ignore[attr-defined]
+        version = cast("AppleVersion", tuple(map(int, release.split(".")[:2])))
+
+    if multiarch is None:
+        multiarch = sys.implementation._multiarch
+    multiarch = multiarch.replace("-", "_")
+
+    ios_platform_template = "ios_{major}_{minor}_{multiarch}"
+
+    # Consider any major.minor version from iOS 12.0 to the version prior to the
+    # version requested by platform. 12.0 is the first iOS version that is known
+    # to have enough features to support CPython. Consider every possible minor
+    # release up to X.9. There highest the minor has ever gone is 8 (14.8 and
+    # 15.8) but having some extra candidates that won't ever match doesn't
+    # really hurt, and it saves us from having to keep an explicit list of known
+    # iOS versions in the code.
+    for major in range(12, version[0]):
+        for minor in range(0, 10):
+            yield ios_platform_template.format(
+                major=major, minor=minor, multiarch=multiarch
+            )
+
+    # Consider every minor version from X.0 to the minor version prior to the
+    # version requested by the platform.
+    for minor in range(0, version[1]):
+        yield ios_platform_template.format(
+            major=version[0], minor=minor, multiarch=multiarch
+        )
+
+    # Consider the actual X.Y version that was requested.
+    yield ios_platform_template.format(
+        major=version[0], minor=version[1], multiarch=multiarch
+    )
```

Comment:
Tags should be yielded in highest to lowest priority order. This feels like this is lowest to highest (i.e. you probably want to reverse the order of the `for` loops and use `range(..., ..., -1)`).